pd/src/info/light_wallet.rs

```diff
@@ -0,0 +1,160 @@
+    light_wallet::{
+        light_wallet_server::LightWallet, AssetListRequest, ChainParamsRequest,
+        CompactBlockRangeRequest, ValidatorInfoRequest,
+    },
+    stake::ValidatorInfo,
+    Protobuf,
+};
+use tonic::Status;
+use tracing::instrument;
+
+// TODO(hdevalence): this still doesn't work, giving up for now
+// We need to use the tracing-futures version of Instrument,
+// because we want to instrument a Stream, and the Stream trait
+// isn't in std, and the tracing::Instrument trait only works with
+// (stable) std types.
+// use tracing_futures::Instrument;
+
+use crate::components::{app::View as _, shielded_pool::View as _, staking::View as _};
+use crate::WriteOverlayExt;
+
+struct WalletOverlay<T: WriteOverlayExt>(T);
+
+#[tonic::async_trait]
+impl<T: WriteOverlayExt> LightWallet for WalletOverlay<T> {
+    type CompactBlockRangeStream =
+        Pin<Box<dyn futures::Stream<Item = Result<CompactBlock, tonic::Status>> + Send>>;
+
+    type ValidatorInfoStream =
+        Pin<Box<dyn futures::Stream<Item = Result<ValidatorInfo, tonic::Status>> + Send>>;
+
+    #[instrument(skip(self, request))]
+    async fn chain_params(
+        &self,
+        request: tonic::Request<ChainParamsRequest>,
+    ) -> Result<tonic::Response<ChainParams>, Status> {
+        self.0.check_chain_id(&request.get_ref().chain_id).await?;
+
+        let chain_params = self
+            .0
+            .get_chain_params()
+            .await
+            .map_err(|_| tonic::Status::unavailable("database error"))?;
+
+        Ok(tonic::Response::new(chain_params.into()))
+    }
+
+    #[instrument(skip(self, request))]
+    async fn asset_list(
+        &self,
+        request: tonic::Request<AssetListRequest>,
+    ) -> Result<tonic::Response<KnownAssets>, Status> {
+        self.0.check_chain_id(&request.get_ref().chain_id).await?;
+
+        let known_assets = self
+            .0
+            .known_assets()
+            .await
+            .map_err(|_| tonic::Status::unavailable("database error"))?;
+        Ok(tonic::Response::new(known_assets.into()))
+    }
+
+    #[instrument(skip(self, request), fields(show_inactive = request.get_ref().show_inactive))]
+    async fn validator_info(
+        &self,
+        request: tonic::Request<ValidatorInfoRequest>,
+    ) -> Result<tonic::Response<Self::ValidatorInfoStream>, Status> {
+        self.0.check_chain_id(&request.get_ref().chain_id).await?;
+
+        let validators = self
+            .0
+            .validator_list()
+            .await
+            .map_err(|_| tonic::Status::unavailable("database error"))?;
+
+        let _show_inactive = request.get_ref().show_inactive;
+        let store = self.0.clone();
+        let s = try_stream! {
+            for validator in validators {
+                let info = store.validator_info(&validator)
+                    .await?
+                    .expect("known validator must be present");
+                // TODO: filter by show_inactive
```

Comment:
Currently, the implementation doesn't respect the `show_inactive` filter, but this doesn't affect the client except to return more data than was desired.